pilot/pkg/bootstrap/server.go

```diff
@@ -198,6 +198,13 @@ func NewServer(args PilotArgs) (*Server, error) {
 	if args.Namespace == "" {
 		args.Namespace = os.Getenv("POD_NAMESPACE")
 	}
+	if args.Config.ClusterRegistriesNamespace == "" {
+		if args.Namespace != "" {
+			args.Config.ClusterRegistriesNamespace = args.Namespace
+		} else {
```

Comment:
One more question, I think we do not need the else now, as 198-199 can always guarantee there is a value for `args.Namespace`

Reply:
I was thinking the same but @sbezverk reminded that there's deployment scenarios with pilot not in a k8s cluster & so it's possible there's no POD_NAMESPACE in env.

Reply:
make sense, do you want to add a comment here to clarify?